File: accsaber-business/src/main/java/de/ixsen/accsaber/business/staff/StaffService.java
package de.ixsen.accsaber.business.staff;

import de.ixsen.accsaber.database.model.staff.StaffUser;
import de.ixsen.accsaber.database.repositories.StaffUserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
public class StaffService {

    private final PasswordEncoder bCryptPasswordEncoder;
    private final StaffUserRepository staffUserRepository;

    @Autowired
    public StaffService(PasswordEncoder bCryptPasswordEncoder, StaffUserRepository staffUserRepository) {
        this.bCryptPasswordEncoder = bCryptPasswordEncoder;
        this.staffUserRepository = staffUserRepository;
    }

    public StaffUser saveNewUser(String username, String password) {
        StaffUser staffUser = new StaffUser();
        staffUser.setUsername(username);
        staffUser.setPassword(this.bCryptPasswordEncoder.encode(password));

        this.staffUserRepository.save(staffUser);
        return staffUser;
    }

    public StaffUser findUserByName(String name) {
        return this.staffUserRepository.findByUsername(name);
    }
}

File: accsaber-business/src/main/kotlin/de/ixsen/accsaber/business/staff/StaffService.kt
package de.ixsen.accsaber.business.staff

import de.ixsen.accsaber.database.model.staff.StaffUser
import de.ixsen.accsaber.database.repositories.model.StaffUserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service

@Service
class StaffService @Autowired constructor(private val bCryptPasswordEncoder: PasswordEncoder, private val staffUserRepository: StaffUserRepository) {
    fun saveNewUser(username: String?, password: String?): StaffUser {
        val staffUser = StaffUser()
        staffUser.username = username
        staffUser.password = bCryptPasswordEncoder.encode(password)
        staffUserRepository.save(staffUser)
        return staffUser
    }
}